python_modules/dagster/dagster/_core/executor/step_delegating/step_delegating_executor.py

```diff
@@ -68,24 +69,44 @@ def __init__(
             ),
         )
         self._should_verify_step = should_verify_step
+
         self._event_cursor: Optional[str] = None
+        self._pop_events_offset = int(os.getenv("EXECUTOR_POP_EVENTS_OFFSET", "0"))
```

Comment:
probably should DAGSTER_ prefix 
"retread" is my current favorite word I have for describing this behavior